src/file_pipe_log/pipe.rs

```diff
@@ -23,14 +23,17 @@ use super::format::{build_recycled_file_name, FileNameExt, LogFileFormat};
 use super::log_file::build_file_reader;
 use super::log_file::{build_file_writer, LogFileWriter};
 
+pub type PathId = usize;
+pub type Paths = Vec<PathBuf>;
+
+/// Main directory path id.
 pub const DEFAULT_PATH_ID: PathId = 0;
+/// Auxiliary directory path id.
+pub const AUXILIARY_PATH_ID: PathId = 1;
```

Comment:
I hold the different perspective on this point.
+ Firstly, the original design to this is to support an **EXTRA** dir to store logs  if and only if the main dir is full, ideally placed on another **DEVICE**. It serves the target that the directory of storing raft logs could be placed on an individual device to make IOs individual to other dirs' IO operations. 
+ Secondly, in normal cases, two directories, specified by `dir` and `auxiliary_dir`, are enough for the practical production envs. And we do not expect to support 3,4, or more dir lists for users.

So, I still recommend to use this design.

Reply:
It is contradicting the whole point of using a vector as paths. If you need to hardcode the path ID, it is clearly better to just use two variables `dir1` and `dir2`, because it checks the code statically instead of runtime (via vector indexing). And my originally idea of using a vector is to not restricting the amount of directories. For example, if in the future we would support #258, user can configure multiple disks to increase the write parallelism.